pkg/internal/apis/certmanager/validation/certificaterequest.go

```diff
@@ -107,6 +113,80 @@ func ValidateCertificateRequestSpec(crSpec *cmapi.CertificateRequestSpec, fldPat
 	return el
 }
 
+// ValidateCertificateRequestApprovalCondition will ensure that only a single
+// 'Approved' or 'Denied' condition may exist, and that they are set to True.
```

Comment:
Is any of this is copied from the Kubernetes CSR validation? If so, it'd probably be worth adding an acknowledgment and a link to the the code we referred to.

If not copied from Kubernetes, then I should go and look and see if they're covering anything that we're not. (or vice versa)

Reply:
Not copied the code per say, just coding the same behaviour. Maybe something I should look at..